src/json-validator.cpp

```diff
@@ -186,64 +186,98 @@ void validate_boolean(const json & /*instance*/, const json &schema, const std::
 	validate_type(schema, "boolean", name);
 }
 
-void validate_numeric(const json &schema, const std::string &name, double value)
+template <class T>
+bool violates_numeric_maximum(T max, T value, bool exclusive)
 {
+	if (exclusive)
+		return value > max;
+
+	return value >= max;
+}
+
+template <class T>
+bool violates_numeric_minimum(T min, T value, bool exclusive)
+{
+	if (exclusive)
+		return value < min;
+
+	return value <= min;
+}
+
+template <class T>
+void validate_numeric(const json &instance, const json &schema, const std::string &name)
```

Comment:
Wondering if I should do a `static_assert` to prevent usage of this template with types other than `int64_t`, `uint64_t`, or `double`. Probably ok since it's a non-public function.